src/shared/common/forms/back-button.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { ApplicationConsumer } from '../../contexts/application.jsx';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+import { withRouter } from 'react-router';
+import {
+    setError,
+    updateApiStatus,
+    clearApiStatus,
+} from '../../contexts/actions/index.jsx';
+import status from '../../helpers/api-status.js';
+
+export class BackButton extends Component {
+
+    handleClick(e) {
+        e.preventDefault();
+        const { dispatch, history } = this.props;
+        const endpoint = `/api/form/case/${this.props.caseId}/stage/${this.props.stageId}/back`;
+        return dispatch(updateApiStatus(status.MOVE_BACK_REQUEST))
+            .then(() => {
+                axios.get(endpoint)
+                    .then(response => {
+                        dispatch(updateApiStatus(status.MOVE_BACK_SUCCESS))
+                            .then(() => {
+                                if (response.data.errors) {
```

Comment:
I think this test should be before setting the success status, which might simplify it a bit